### PR TITLE
chore(STONEINTG-867): enable vendoring

### PR DESCRIPTION
**Pull request by MartinBasti**
Enable vendoring to prevent issues when a repo disappear

## Maintainers will complete the following section

- [ ] Commit messages are descriptive enough ([hints](https://www.freecodecamp.org/news/how-to-write-better-git-commit-messages/))
- [ ] Code coverage from testing does not decrease and new code is covered ([check the PR coverage on codecov](https://app.codecov.io/gh/redhat-appstudio/integration-service/pulls))
- [ ] [Controllers diagrams](https://github.com/redhat-appstudio/integration-service/tree/main/docs) are updated when PR changes controllers code  (if applicable)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -15,9 +15,6 @@ testbin/*
 *.out
 cover.html
 
-# folder generated by go mod vendor
```
Comment (MartinBasti) [inline .gitignore:18]: This is important change

File: README.md [hunk 2]
```diff
@@ -37,6 +37,19 @@ $ IMG=quay.io/user/integration-service:my-tag make img-build
 $ IMG=quay.io/user/integration-service:my-tag make img-push
 ```
 
+### Adding/updating a dependency
```
Comment (MartinBasti) [inline README.md:40]: This is important change, everything else is just magic

**Comment by MartinBasti** [comment]
Linter fails due this https://github.com/redhat-appstudio/integration-service/pull/664

**Comment by MartinBasti** [comment]
I cannot see why code/snyk is failing

**Review by hongweiliu17 — APPROVED** [review]
LGTM

**Comment by MartinBasti** [comment]
There are ginkgo-linter issues, but I haven't changed tests, it's supposed to detect only new changes, maybe due the size of this PR, detection of only new issues is broken. I'd waive this check

**Review by dirgim — APPROVED** [review]
LGTM

**Pull request merged**